app/__tests__/AddRuleModal.test.jsx

```diff
@@ -0,0 +1,62 @@
+import * as React from "react";
+import { shallow } from "enzyme";
+import { AddRuleModal } from "./../components/AddRuleModal";
+
+const createTestProps = props => ({
+  handleClose: jest.fn(),
+  showModal: false,
+  modalMethod: "GET",
+  addRequest: jest.fn(),
+  modalUrl: "www.codemancers.com",
+  updateModalMethod: jest.fn(),
+  updateModalUrl: jest.fn()
+  // allow to override common props
+  ...props
+});
+
+describe("AddRuleModal component test", () => {
+  const wrapper, props;
+  beforeEach(() => {
+    jest.clearAllMocks();
+    props = createTestProps();
+    wrapper = shallow(<AddRuleModal {...props} />);
+  });
+
+  describe("default", () => {
+    test("Contains a Modal HOC component", () => {
+      expect(wrapper).toBeDefined();
+      expect(wrapper.find("Modal")).toHaveLength(1);
+    });
+    test("Contains a input, select and a button element", () => {
+      expect(wrapper.find("input")).toBeTruthy();
```

Comment:
AFAIK, this won't work. `wrapper.find("input")` always returns an object. so we have to check for the length to make sure the element exists